app/models/vm_or_template.rb

```diff
@@ -472,6 +472,60 @@ def self.invoke_task_local(task, vm, options, args)
     )
   end
 
+  def self.invoke_tasks_remote_queue(options, deliver_on = nil)
+    queue_opts = {
+      :role        => "user_interface",
+      :class_name  => base_class.name,
+      :method_name => 'invoke_tasks_remote',
+      :args        => [options]
+    }
+    queue_opts[:deliver_on] = deliver_on if deliver_on
+    MiqQueue.put(queue_opts)
+  end
+
+  def self.invoke_api_task(hostname, action, ids)
+    require 'rest-client'
+    require 'json'
+
+    api_user     = Settings.webservices.remote_miq_api.user
+    api_password = MiqPassword.try_decrypt(Settings.webservices.remote_miq_api.password)
```

Comment:
I think I may remove the default password because then we are exposing the value both encrypted and decrypted.
I'll put a check in for that.